yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -24,22 +52,47 @@ export class Archiver implements L2BlockSource, UnverifiedDataSource {
    */
   private unverifiedData: UnverifiedData[] = [];
 
-  private unwatchBlocks: (() => void) | undefined;
-  private unwatchUnverifiedData: (() => void) | undefined;
+  /**
+   * Next L1 block number to fetch `L2BlockProcessed` logs from (i.e. `fromBlock` in eth_getLogs).
+   */
+  private nextL2BlockFromBlock = 0n;
+
+  /**
+   * Next L1 block number to fetch `UnverifiedData` logs from (i.e. `fromBlock` in eth_getLogs)
+   */
+  private nextUnverifiedDataFromBlock = 0n;
+
+  /**
+   * The time after which we will consider the current block number to be outdated.
+   * Note: 12 seconds may seem high but in this case it doesn't matter because we want to introduce a delay
```

Comment:
There are parts that seems overly complicated, certainly for our current use case. Re-orgs aren't necessarily a problem. If sequencer selection is deterministic (and this is some way off yet) then I don't think we particularly need to worry. And we certainly don't need to be concerned about this currently.

Reply:
So would you prefer to not deal with the confirmations at all now?

Reply:
I don't think we need to worry about confirmations for the time being.